MyFTP/ProgramClient/Client.cs

```diff
@@ -0,0 +1,74 @@
+namespace ProgramClient;
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Net;
+using System.Net.Sockets;
+using System.Threading;
+using System.Threading.Tasks;
+
+/// <summary>
+/// Class for MyFTP client
+/// </summary>
+public class Client
+{
+    private readonly IPAddress _ip;
+    private readonly int _port;
+
+    /// <summary>
+    /// Constructor of client
+    /// </summary>
+    public Client(IPAddress ip, int port)
+    {
+        _ip = ip;
+        _port = port;
+    }
+    
+    /// <summary>
+    /// Returns files and directories on server
+    /// </summary>
+    public async Task<List<(string name, bool isDir)>> List(string path, CancellationToken token)
+    {
+        using var client = new TcpClient();
+        await client.ConnectAsync(_ip, _port, token);
+        await using var stream = client.GetStream();
+        await using var writer = new StreamWriter(stream){AutoFlush = true};
+        using var reader = new StreamReader(stream);
+        await writer.WriteLineAsync("1 " + path);
+        var files = new List<(string, bool)>();
+        var response = await reader.ReadLineAsync();
+        if (response == null) throw new NullReferenceException();
```

Comment:
```suggestion
        if (response == null) 
        {
            throw new NullReferenceException();
        }
```
А ещё NullReferenceException никогда нельзя бросать (и ловить), это "системное" исключение (зарезервированное для виртуальной машины .NET). Бросать и ловить нельзя, чтобы если оно таки возникло, это бы означало сразу ошибку программирования, а не что-то ещё.

Тут надо бросать своё какое-нибудь исключение, которое поясняет, что именно произошло.